### ISSUE TITLE
Accessibility Challenge in 'Single Selection with Option Group': Inability to Type and Select Matching Option

### ISSUE THREAD

**Issue by zombaki** [bithost-gmbh/ngx-mat-select-search#448]
**Describe the bug**
This accessibility issue pertains to the "Single Selection with Option Group" feature, where users are facing a notable challenge. Specifically, users find themselves unable to input text and select the corresponding option effectively. This limitation hinders the user experience, as the expected functionality of typing and seamlessly entering the matching option is compromised

**To Reproduce**
1 . Go to search input field for Option Group
2. Type word which you want to be filtered 
3. Press Enter key so that is selected

**Expected behavior**
Searched element should be selected and populated in the input field

**Screenshots**

**Desktop (please complete the following information):**
Can be replicated in stackblitz 

**Smartphone (please complete the following information):**
NA

**Comment by macjohnny** [comment]
@zombaki thanks for reporting! Do you want to propose a fix?
